sympy/combinatorics/subsets.py

```diff
@@ -59,11 +58,26 @@ def __new__(cls, subset, superset):
             if elem not in superset:
                 raise ValueError('The superset provided is invalid as it does '
                                  'not contain the element {}'.format(elem))
-        obj = Atom.__new__(cls)
+        obj = object.__new__(cls)
         obj._subset = subset
         obj._superset = superset
         return obj
 
+    def __eq__(self, other):
+        """Return a boolean indicating whether a == b on the basis of
+        whether both objects are of the class Subset and if the values 
+        of the subset and superset attributes are the same.
+        """
+        if not isinstance(self, Subset):
+            return False
+        if not isinstance(other, Subset):
+            return False
```

Comment:
You don't need to check the type of `self` here. Also I would return NotImplemented rather than False:
```python
if not isinstance(other, Subset):
    return NotImplemented
return self.subset == other.subset and self.superset == other.superset
```